sympy/polys/polytools.py

```diff
@@ -3773,6 +3775,57 @@ def nth_power_roots_poly(f, n):
 
         return r.replace(t, x)
 
+    def same_root(f, a, b):
+        """
+        Decide whether two roots of this polynomial are equal.
+
+        Examples
+        ========
+
+        >>> from sympy import Poly, cyclotomic_poly, exp, I, pi
+        >>> f = Poly(cyclotomic_poly(5))
+        >>> r0 = exp(2*I*pi/5)
+        >>> indices = [i for i, r in enumerate(f.all_roots()) if f.same_root(r, r0)]
+        >>> print(indices)
+        [3]
+
+        Raises
+        ======
+
+        DomainError
+            If the domain of the polynomial is not :ref:`ZZ`, :ref:`QQ`,
+            :ref:`RR`, or :ref:`CC`.
+        MultivariatePolynomialError
+            If the polynomial is not univariate.
+        PolynomialError
+            If the polynomial is of degree < 2.
+
+        """
+        if f.is_multivariate:
+            raise MultivariatePolynomialError(
+                "Must be a univariate polynomial")
+
+        dom_delta_sq = f.rep.mignotte_sep_bound_squared()
+        delta_sq = f.domain.get_field().to_sympy(dom_delta_sq)
+        # We have delta_sq = delta**2, where delta is a lower bound on the
+        # minimum separation between any two roots of this polynomial.
+        # Let eps = delta/3, and define eps_sq = eps**2 = delta**2/9.
+        eps_sq = delta_sq / 9
+
+        r, _, _, _ = evalf(1/eps_sq, 1, {})
+        n = fastlog(r)
+        # Then 2^n > 1/eps**2.
+        m = math.ceil(n/2)
```

Comment:
Maybe I'm overthinking this but at first glance this seems an unnecessary use of machine precision floating point...

Reply:
If `n` is guaranteed to be an an integer then I guess this would work:
```
m = (n // 2) + (n % 2)
```
(Add zero if n is even or 1 if n is odd)

Reply:
Interesting. Would not have thought about the efficiency here. Will change it.

Reply:
It's not so much efficiency as correctness:
```python
In [46]: e = 2 ** 53 + 1

In [47]: math.ceil(e / 2)
Out[47]: 4503599627370496

In [48]: (e // 2) + (e % 2)
Out[48]: 4503599627370497
```
That's probably way outside the range of what is possible here but in general we should avoid trusting floating point calculations unless necessary.